lib/frisby.js

```diff
@@ -1217,7 +1217,7 @@ function _jsonContains(jsonBody, jsonTest) {
         var keyType = jsonTest[key].prototype;
 
         // User-supplied callback (anonymous function)
-        if(_toType(keyType) === "object" && arguments.callee.caller.name === "") {
+        if(_toType(keyType) === "object" && (arguments.callee.caller.name === "" || arguments.callee.caller.name === arguments.callee.name)) {
```

Comment:
agreed; the original problem I was fixing here was that the comparison to empty string broke recursion, which resulted in deep objects not being compared. I verified that the changes on master pass the (non-array) tests in this PR, so this change should now be unnecessary.